docs/generator_template.md

```diff
@@ -0,0 +1,569 @@
+
+```js
+// 1
+import { File, Text } from '@asyncapi/generator-react-sdk'
+export default function ({ asyncapi, params }) {
+  return (
+    <File name="client.py">
+    // 2
+      <Text newLines={2}>import paho.mqtt.client as mqtt</Text>
+    // 3
+      <Text newLines={2}>mqttBroker = "{asyncapi.servers().get(params.server).url()}"</Text>
+    // 4
+      <Text newLines={2}>class {asyncapi.info().title().replaceAll(' ', '')}Client:</Text>
+    // 5
+      <Text indent={2} newLines={2}>
+        {`def __init__(self):
+            self.client = mqtt.Client()
+            self.client.connect(mqttBroker)`}
+      </Text>
+      </File>
+  )
+}
+```
+
+1. Import the **Text** component that will wrap strings so they are indented properly in the output. Your import statement should now look like this: `import { File, Text } from '@asyncapi/generator-react-sdk'`.
+2. When the paho module import is rendered in **client.py** file, it will add two extra new lines.
+3. The broker url is templated in a Text component removing the `$` from the string template.
+4. Dynamically get the class name **TemperatureServiceClient** from the AsyncAPI document from the **info** object using the Parser API using the code: `asyncapi.info().title()` . It will return `Temperature Service`, then remove the spaces and add Client as a suffix.
+5. There is no templating needed in the `__init__` function, there is only hardcoded information.
+
+> If you're on the fence about which templating engine you should use in your template, check out the [React render engine](https://www.asyncapi.com/docs/tools/generator/react-render-engine) and [nunjucks render engine](https://www.asyncapi.com/docs/tools/generator/nunjucks-render-engine) documentation.
+In the next section, you'll refactor your template to use React.
+
+### Creating a reusable component
+
+If you had two [channels](https://www.asyncapi.com/docs/concepts/channel), one to watch if the temperature drop below 22 °C and one to check if the temperature is above 22 °C, the generated output code would look like this:
+
+```python
+import paho.mqtt.client as mqtt
+
+mqttBroker = "test.mosquitto.org"
+
+class TemperatureServiceClient:
+
+  def __init__(self):
+              self.client = mqtt.Client()
+              self.client.connect(mqttBroker)
+
+  def sendTemperatureDrop(self, id):
+          topic = "temperature/droped"
+          self.client.publish(topic, id)
+  def sendTemperatureRise(self, id):
+          topic = "temperature/risen"
+          self.client.publish(topic, id)
+
+```
+
+You'll then need to template to dynamically generate `sendTemperatureDropped` and `sendTemperatureRisen` functions in the generated code based off the AsyncAPI document content. The goal is to write template code that returns functions for channels that the Temperature Service application is subscribed to. The template code to generate these functions will look like this:
```

Comment:
names of functions do not match the code

Reply:
Nice catch